problem_builder/public/js/student_answers_dashboard.js

```diff
@@ -0,0 +1,299 @@
+function StudentAnswersDashboardBlock(runtime, element) {
+    'use strict';
+    var $element = $(element);
+
+    // Pagination
+
+    var Result = Backbone.Model.extend({
+
+        initialize: function(attrs, options) {
+            _.each(_.zip(Result.properties, options.values), function(pair) {
+                this.set(pair[0], pair[1]);
+            }, this);
+        }
+
+    }, { properties: ['section', 'subsection', 'unit', 'type', 'question', 'answer', 'username'] });
+
+    var Results = Backbone.PageableCollection.extend({
+
+        model: Result,
+
+        getCurrentPage: function(object) {
+            var currentPage = this.state.currentPage;
+            if (object) {
+                return this.getPage(currentPage);
+            }
+            return currentPage;
+        },
+
+        getTotalPages: function() {
+            return this.state.totalPages;
+        }
+
+    });
+
+    var ResultsView = Backbone.View.extend({
+
+        render: function() {
+            this._insertRecords(this.collection.getCurrentPage(true));
+            this._updateControls();
+            this.$('#total-pages').text(this.collection.getTotalPages() || 0);
+            return this;
+        },
+
+        _insertRecords: function(records) {
+            var tbody = this.$('tbody');
+            tbody.empty();
+            records.each(function(result, index) {
+                var row = $('<tr>');
+                if (index % 2 === 0) {
+                    row.addClass('even');
+                }
+                _.each(Result.properties, function(name) {
+                    row.append($('<td>').text(result.get(name)));
+                });
+                tbody.append(row);
+            }, this);
+            if (this.collection.getTotalPages()) {
+                this.$('#current-page').text(this.collection.getCurrentPage());
+            } else {
+                this.$('#current-page').text(0);
+            }
+        },
+
+        events: {
+            'click #first-page': '_firstPage',
+            'click #prev-page': '_prevPage',
+            'click #next-page': '_nextPage',
+            'click #last-page': '_lastPage'
+        },
+
+        _firstPage: function() {
```

Comment:
@bradenmacdonald That's an interesting question! :) I guess I tend think of the `events` hash as the public API for a `Backbone.View`. So if client code wants to effect a change in a view in most cases I would expect it to trigger an appropriate event (instead of calling one of its methods). That's why I added a leading `_` to these handlers.

But maybe that's just me -- if you disagree I'll remove the underscores :)